cmd/execution/main.go

```diff
@@ -564,6 +570,83 @@ func main() {
 		}).Run()
 }
 
+// Get and log the epoch counters from the protocol state and from the FlowEpoch smart contract.
+func logEpochNumber(vm *fvm.VirtualMachine, vmCtx fvm.Context, executionState state.ExecutionState, node *cmd.NodeConfig, log zerolog.Logger) {
+	logFail := func(err error) {
+		log.Error().Err(err).Msg("could not log epoch number")
+	}
+
+	// Get the epoch counter form the protocol state
+	protocolStateEpoch, err := node.State.
+		AtBlockID(node.RootBlock.ID()).
```

Comment:
```suggestion
		Final().
```

`AtBlockID(node.RootBlock.ID())` will reference the root block of the snapshot, `Final()` will reference the latest finalized block (staying up to date as the execution node incorporates blocks). I'm pretty sure we want the up-to-date version here, since we're using the latest executed block for reading from the smart contract.